### PR TITLE
Feat add auth

### PR DESCRIPTION
**Pull request by klemby**

### CODE REVIEW

File: trickster/endpoints/mocked.py [hunk 1]
```diff
@@ -20,9 +20,13 @@ def mocked_response(request: Request, mocked_router: Router = Depends(get_router
...
     if match := mocked_router.match(request):  # noqa: SIM102 - nested if statements
-        if matched_response := match.route.get_response(match):
-            match.route.hits += 1
-            response = matched_response
+        try:
+            match.route.authenticate(request)
+            if matched_response := match.route.get_response(match):
+                match.route.hits += 1
+                response = matched_response
+        except HTTPException:
```
Comment (pklejch) [inline trickster/endpoints/mocked.py:28]: As Jakub already told you, you are converting any HTTPException to 401. Throw some your explicit exception `AuthenticationError` and translate it to 401.

Comment (klemby) [reply trickster/endpoints/mocked.py:28]: This issue is fixed here
https://github.com/TricksterOrg/trickster/pull/29/files#

File: trickster/model.py [hunk 2]
```diff
@@ -251,6 +307,7 @@ class Route(BaseModel):
     responses: list[Response] = Field(default_factory=list, description='Possible responses of the route')
     response_selector: ResponseSelector = Field(
         default=ResponseSelector.RANDOM, description='Strategy for response selection')
+    auth: Union[Auth.get_subclasses()] | None = Field(default=None, discriminator='method')  # type: ignore
```
Comment (pklejch) [inline trickster/model.py:310]: This is nice and all, but if it doesn't work with mypy, I would personally rather explicitly fill all valid options `Union[TokenAuth, XYZAuth,]` and let the mypy be happy. But wait for others what they thinking of this...

Comment (JakubTesarek) [reply trickster/model.py:310]: I recommended this to Erik as a solution. I see your point but the goal was also that Route doesn't have to have the responsibility of knowing what are the possible options. One solution might be declaring this type just bellow the Auth class like this:

`auth_types = cast(Auth | None, Union[Auth.get_subclasses()] | None)`

Or something like this. It should make mypy happy, pydantic happy and keep the responsibility of listing auth types outside.

Comment (JakubTesarek) [reply trickster/model.py:310]: Or we can make the method return the typing type and not tuple. But that would require renaming it and making it very clear, what it's for 

I would prefer this solution and rather disable mypy here.

**Review by JakubTesarek — APPROVED** [review]

**Pull request merged**